src/context.ts

```diff
@@ -0,0 +1,30 @@
+import { Db } from 'mongodb'
+import { createSubmissionsRepository } from './submission/mongoRepository'
+import { createSubmissionService } from './submission/service'
+import { createSchemaRepository } from './schema/mongoRepository'
+import { createSchemaService } from './schema/service'
+import { createEntityRepository } from './entity/mongoRepository'
+import { createEntityService } from './entity/service'
+
+interface IO {
+  database: Db
+}
+
+export async function createAppContext(io: IO) {
+  const { database } = io
```

Comment:
Why not just pass the `database` directly? Do we expect more IO params in the future?

Reply:
Yes. Honestly, I wasn't sure about this step but it seemed a bit weird to pass it just as one parameter. We might not need to add more deps like this in the near future but if we do we should not add it as another parameter but pass it as a property of one deps object... the naming `io` was just first idea that came to my mind, maybe not so great 🤷‍♂️ but it kind of make sense to me.